### PR TITLE
sys: Fixes for MinGW support

### PR DESCRIPTION
**Pull request by akien-mga**
Tested successfully in Godot to compile with MinGW-GCC and MinGW-Clang
for Windows.

This doesn't include needed changes to CMakeLists.txt to support using
a MinGW-GCC based toolchain to compile Embree with CMake.

Co-authored-by: @RandomShaper

Related to #279.

### CODE REVIEW

File: common/sys/platform.h [hunk 1]
```diff
@@ -99,7 +99,7 @@
 #define dll_import 
 #endif
 
-#ifdef __WIN32__
+#if defined(__WIN32__) && !defined(__MINGW32__)
```
Comment (akien-mga) [inline common/sys/platform.h:102]: The same might be relevant on line 94 for the definitions of `dll_export` and `dll_import`. But it's not part of the patch we're using in production for Godot (https://github.com/godotengine/godot/blob/master/thirdparty/embree/patches/godot-changes-misc.patch), so for this PR I chose to play it safe and only include what we tested.

Comment (akien-mga) [inline common/sys/platform.h:102]: This was needed to fix a crash issue when compiling Embree with MinGW-GCC, which we experienced upstream: https://github.com/godotengine/godot/issues/45097

**Comment by svenwoop** [comment]
I merged this into our internal devel branch. The fix will be in the next release.
